.gitignore

```diff
@@ -1,21 +1,167 @@
-**/__pycache__
-.pytest_cache
-.benchmarks
-**egg-info
-**/.DS_Store
-**/build
-dist/*
-docs/source/gallery/**
-docs/source/benchmarks/**
```

Comment:
why not keep them? they're rather useful.

Reply:
> ```gitignore
> **/__pycache__
> .pytest_cache
> **egg-info
> **/build
> dist/*
> docs/source/gallery/**
> docs/source/benchmarks/**
> ```

These entries are still included in the new file.

> ```gitignore
> .benchmarks
> ```

For this one, I'm not sure why this is necessary.

Do you remember why this was added?

> ```gitignore
> **/.DS_Store
> ```

This entry is specific to MacOS: as `.gitignore` should only contain what we want *every* contributors to ignore, it's not appropriate to include this line.

Note that this also applies to `.vscode/` or similar editor configs.

Reply:
Resolved in bc14615dc405e1f8b55fb32b4687bcc602996096 .